packages/inter-protocol/src/auction/sortedOffers.js

```diff
@@ -0,0 +1,144 @@
+import { makeRatio } from '@agoric/zoe/src/contractSupport/index.js';
+import { TimeMath } from '@agoric/time';
+
+import { decodeNumber, encodeNumber } from '../vaultFactory/storeUtils.js';
+
+// We want earlier times to sort the same direction as higher prices, so we
+// subtract the timestamp from millisecond time in the year 2286. This works for
+// timestamps in seconds or millis. The alternative considered was inverting,
+// but floats don't have enough resolution to convert back to the same timestamp
+// This will work just fine for at least 250 years. And notice that these
+// timestamps are used for sorting during an auction and don't need to be stored
+// long-term. We could safely subtract from a timestamp that's now + 1 month.
+const FarFuture = 10000000000000n;
+const encodeTimestamp = t => FarFuture - t;
```

Comment:
please type `t` as we have two kinds of timestamps now. Ideally this takes `TimestampRecord`